duckduckgo/iOS
# adblock usage fixes

Reviewer: Mia or Caine
Asana: https://app.asana.com/0/414235014887631/476166083909434
CC:

**Description**:

Fixes two problems reconstructing the cached adblock structures:

* Using the wrong data structure to reconstruct the exception bloom filter
* Re-create Set objects on the raw filters

**Steps to test this PR**:

1. Visit evanscycles.com - observe trackers blocked (quite a few) and note that https://cdn.tagcommander.com/1705/tc_abtest.js is not blocked
1. Refresh the page and check again (using cached version)
1. Browse a few sites and confirm trackers blocked and privacy grade look reasonable

Additional tests to check for regressions:

***Test 1***:

1. Clean launch of the app
1. Go directly to huffingtonpost.com
1. Perform a search: DDG Grade should be A

Repeat with privacy protection off

***Test 2***:

1. Clear data
1. Search for cnn
1. Navigate to cnn.com 
1. Swipe back to DDG
1. DDG should remain an A

(repeat a few times to be sure)

***Test 3***

1. Go to cnn.com directly
1. Go to ddg.gg directly 
1. Redirects to DuckDuckGo.com which has an A grade
